### PR TITLE
Handle ESC key in game selection UI

### PR DESCRIPTION
**Pull request by seanshpark**
This will add ESC key to cancel operation in game selection UI
like clicking on mouse right button

### CODE REVIEW

**Comment by seanshpark** [comment]
@wateret , I'm not sure this patch will work without problems. Please add comments if there are any.
- I've added codes to call `gv_->PopUIState();` by looking in `state_ui.cc` with `OnMouseButtonEvent` method that also calls `PopUIState()`

File: game/src/gui/app/state_ui.cc [hunk 1]
```diff
@@ -352,6 +352,19 @@ bool StateUIUnitSelected::OnMouseButtonEvent(const foundation::MouseButtonEvent&
   return true;
 }
 
+bool StateUIUnitSelected::OnKeyEvent(const foundation::KeyEvent& e) {
+  if (e.IsKeyUp()) {
```
Comment (wateret) [inline game/src/gui/app/state_ui.cc]: Could you please add a line comment like `TODO Design a way to remap the key`? Not only this place but other OnKeyEvent handlers.

File: game/src/gui/foundation/key_event.h [hunk 2]
```diff
@@ -11,6 +11,10 @@ class KeyEvent {
  public:
   enum class State : uint16_t { kIdle, kDown, kUp };
   enum class Mod : uint16_t { kNone = 0, kCtrl = 1, kShift = 2, kAlt = 4 };
+  enum {
```
Comment (wateret) [inline game/src/gui/foundation/key_event.h]: Please make it to be `enum class Key`. The name also can be something better than just `Key`.

File: game/src/gui/foundation/key_event.h [hunk 3]
```diff
@@ -11,6 +11,10 @@ class KeyEvent {
...
+    ESC = 27,
```
Comment (wateret) [inline game/src/gui/foundation/key_event.h]: According to our enum rules, this should be `kEsc`.

File: game/src/gui/app/state_ui.cc (hunk 1 above)
Comment (seanshpark) [reply game/src/gui/app/state_ui.cc]: OK

File: game/src/gui/foundation/key_event.h (hunk 3 above)
Comment (seanshpark) [reply game/src/gui/foundation/key_event.h]: OK

File: game/src/gui/foundation/key_event.h (hunk 2 above)
Comment (seanshpark) [reply game/src/gui/foundation/key_event.h]: Had compile error when used with `enum class Code { ESC=27; };`
```
    switch (e.GetCode()) {
      case foundation::KeyEvent::ESC:
```
Do you know how to solve this?

**Pull request merged**